relay-sampling/src/dsc.rs

```diff
@@ -149,13 +149,24 @@ mod sample_rate_as_string {
     where
         D: serde::Deserializer<'de>,
     {
-        let value = match Option::<Cow<'_, str>>::deserialize(deserializer)? {
+        #[derive(Debug, Clone, Deserialize)]
+        #[serde(untagged)]
+        enum StringOrFloat<'a> {
+            String(Cow<'a, str>),
+            Float(f64),
+        }
+
+        let value = match Option::<StringOrFloat>::deserialize(deserializer)? {
             Some(value) => value,
             None => return Ok(None),
         };
 
-        let parsed_value =
-            serde_json::from_str(&value).map_err(|e| serde::de::Error::custom(e.to_string()))?;
+        let parsed_value = match value {
+            StringOrFloat::Float(f) => f,
+            StringOrFloat::String(s) => {
+                serde_json::from_str(&s).map_err(|e| serde::de::Error::custom(e.to_string()))?
```

Comment:
Untested, but I think this should work:

```suggestion
                serde_json::from_str(&s).map_err(serde::de::Error::custom)?
```